### PR TITLE
[WIP] adding ubuntu 18.04 bionic support 

### PR DESCRIPTION
**Pull request by mac-arrap**
This should work but I don't know if everything I configured is correct or if there are endpoints for artifact to be stored.

Signed-off-by: Camilo Parra <caeparra@gmail.com>

### CODE REVIEW

**Comment by hyperledger-github** [comment]
Could one of the committers please verify this patch?

**Comment by ashcherbakov** [comment]
@mac-arrap 
Thanks for doing this
1) The changes need to be done in Plenum as well, not only in Node
2) 18.04 ubuntu support is not only about build scripts. This is also about 
 - Python 3.6 support
 - CI (running tests on Ubuntu 18.04 to make sure that it works there)
 - dependency management (we use strict pins for dependencies and some of them match canonical versions for Ubuntu 16.04; we need to figure out how to organize it for 18.04)

Please consider writing a PoA for the whole epic. You can use https://jira.hyperledger.org/browse/INDY-2186 epic, it already has some thinking and a rough plan there.

File: build-scripts/ubuntu-1804/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+FROM ubuntu:18.04
+
+RUN apt-get update -y && apt-get install -y \
+    # common stuff
+        git \
+        wget \
+        unzip \
+        python3.5 \
```
Comment (ashcherbakov) [inline build-scripts/ubuntu-1804/Dockerfile]: Ubuntu 18.04 should work with Python 3.6 which is a default there

File: build-scripts/ubuntu-1804/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+### Build indy-node using docker
+
+```
+./build-indy-node-docker.sh <path-to-sources> <version>
+```
+Built package is placed in a docker volume `indy-node-deb-u1604`.
```
Comment (ashcherbakov) [inline build-scripts/ubuntu-1804/README.md]: Should it be `indy-node-deb-u1804`?

File: build-scripts/ubuntu-1804/build-indy-node.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+PACKAGE_VERSION=${4:-$VERSION}
+
+PACKAGE_NAME=indy-node
+
+# copy the sources to a temporary folder
+TMP_DIR="$(mktemp -d)"
+cp -r "${INPUT_PATH}/." "${TMP_DIR}"
+
+# prepare the sources
+cd "${TMP_DIR}/build-scripts/ubuntu-1604"
```
Comment (ashcherbakov) [inline build-scripts/ubuntu-1804/build-indy-node.sh]: Should it be `build-scripts/ubuntu-1804`?

**Comment by mac-arrap** [comment]
@ashcherbakov I have added some more items to make the CI pipeline work. Could you tell me which items are specific to 16.04? I have changed libsodium18 to libsodium23  for 18.04 support

**Comment by Shubham-koli** [comment]
Ubuntu 18.04 supported?

**Comment by ashcherbakov** [comment]
@Shubham-koli 
No, the PR hasn't been merged yet. Only Ubuntu 16.04 is officially supported. 
But we have plans to support Ubuntu 20.04 (skipping 18.04).

**Comment by brentzundel** [comment]
Next steps: Finish work on complementary Plenum PR https://github.com/hyperledger/indy-plenum/pull/1332
Pipeline also needs to be modified so that CI can handle testing both Ubuntu 16 and Ubuntu 18.

**Comment by mac-arrap** [comment]
Anyone working on 20.04 fixes this would get you started on what you need to fix.

**Comment by WadeBarnes** [comment]
(ci) test this please

**Comment by WadeBarnes** [comment]
@mac-arrap, We're getting the following errors in the build process, any thoughts?

image

image

**Comment by WadeBarnes** [comment]
Likely because it's not using the new/updated pipeline files.

**Comment by mac-arrap** [comment]
@WadeBarnes so it looks like it's looking for a file that was renamed. That new file is now called `ci/ubuntu-1604.dockerfile`

**Comment by WadeBarnes** [comment]
(ci) test this please

**Comment by cjhowland** [comment]
@mac-arrap Could you provide a brief update on this PR? With recent branch changes and the current development process on Indy Node/Plenum, we are wondering if these changes are still viable. If so, please close this PR and retarget it to `main`.

We will discuss and determine which PRs should be merged or closed at the next Indy Contributors Working Group meeting on May 9, 2023.

**Comment by pSchlarb** [comment]
I think it is save to say that this is outdated, since the branch switch to Ubuntu 20 already happened. So I would close this pr. If anything important was missed, please feel free to comment.

**Pull request closed without merging**